webapi/src/main/scala/org/knora/webapi/responders/v2/StandoffResponderV2.scala

```diff
@@ -79,63 +80,63 @@ class StandoffResponderV2 extends Responder {
 
-            textLocation: LocationV1 = textRepresentationResponse match {
-                case textRepr: ResourceFullResponseV1 if textRepr.resinfo.isDefined && textRepr.resinfo.get.restype_id == OntologyConstants.KnoraBase.XSLTransformation =>
-                    val locations: Seq[LocationV1] = textRepr.resinfo.get.locations.getOrElse(throw BadRequestException(s"no location given for $xslTransformationIri"))
-
-                    locations.headOption.getOrElse(throw BadRequestException(s"no location given for $xslTransformationIri"))
-
-                case other => throw BadRequestException(s"$xslTransformationIri is not an ${OntologyConstants.KnoraBase.XSLTransformation}")
+            xsltFileValue: TextFileValueContentV2 = textRepresentationResponseV2.resources.headOption match {
+                case Some(resource: ReadResourceV2) if resource.resourceClass == OntologyConstants.KnoraBase.XSLTransformation => resource.values.get(OntologyConstants.KnoraBase.HasTextFileValue) match {
+                    case Some(values: Seq[ReadValueV2]) if values.size == 1 => values.head match {
+                        case value: ReadValueV2 => value.valueContent match {
+                            case textRepr: TextFileValueContentV2 => textRepr
+                            case other => throw InconsistentTriplestoreDataException(s"${OntologyConstants.KnoraBase.XSLTransformation} $xslTransformationIri is supposed to have exactly one value of type ${OntologyConstants.KnoraBase.TextFileValue}")
+                        }
+                    }
+                    case None => throw InconsistentTriplestoreDataException(s"${OntologyConstants.KnoraBase.XSLTransformation} has no property ${OntologyConstants.KnoraBase.HasTextFileValue}")
+                }
+                case None => throw BadRequestException(s"Resource $xslTransformationIri is not a ${OntologyConstants.KnoraBase.XSLTransformation}")
             }
 
-            // check if `textLocation` represents an XSL transformation
-            _ = if (!(textLocation.format_name == "XML" && textLocation.origname.endsWith(".xsl"))) {
-                throw BadRequestException(s"$xslTransformationIri does not have a file value referring to a XSL transformation")
+            // check if `xsltFileValue` represents an XSL transformation
+            _ = if (!(xsltFileValue.internalMimeType == "text/xml" && xsltFileValue.originalFilename.endsWith(".xsl"))) {
+                throw BadRequestException(s"$xslTransformationIri does not have a file value referring to an XSL transformation")
             }
-        } yield textLocation
 
-        val recoveredTextLocationFuture = textLocationFuture.recover {
+            xsltUrl: String = s"${settings.internalSipiFileServerGetUrl}/${xsltFileValue.internalFilename}"
```

Comment:
@subotic Is this Sipi URL correct? 

Using the v2 responder, I could get rid of the URL replacing operation.